torchrl/envs/libs/pettingzoo.py

```diff
@@ -498,26 +514,8 @@ def _reset(
 
         return tensordict_out
 
-    def _reset_aec(self, tensordict=None, **kwargs) -> Tuple[Dict, Dict]:
-        all_done = True
-        if tensordict is not None:
-            _resets = []
-            for done_key in self.done_keys:
-                _reset_key = _replace_last(done_key, "_reset")
-                _reset = tensordict.get(_reset_key, default=None)
-                if _reset is None:
-                    continue
-                _resets.append(_reset)
-            if len(_resets) < len(self.done_keys):
-                all_done = False
-            else:
-                for _reset in _resets:
-                    if not _reset.all():
-                        all_done = False
-                        break
-
-        if all_done:
-            self._env.reset(**kwargs)
+    def _reset_aec(self, **kwargs) -> Tuple[Dict, Dict]:
+        self._env.reset(**kwargs)
```

Comment:
You can check but I think so yes

Reply:
to me it seems that if we keep the lower level _resets than this will be called when any of those is true, even if the higher reset will be set with a `.all()` policy

Reply:
sorry I don't get that comment. Do you mean that `def _reset` should read `"_reset"` from the input tensordict, and call `_reset_aec` and `_reset_parallel` only if ``True`` / absent?